fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -1112,13 +1132,18 @@ impl<'vm, S> CodeRootCtx<'vm, S> {
             self.gas_cost,
             len as u64,
         )?;
-        let root = self
-            .storage
-            .storage_contract(&contract_id)
+        // TODO: Can we write this to the stack and avoid allocation?
```

Comment:
Since we don't know `len` statically I don't think we can put this on the stack, unless we want to over-allocate if we have a upper bound of the contract size we can use.

```suggestion
```

Reply:
I remember discussing this with @xgreenx during one of the weekly syncs. 
I think the only optimization we can do at this stage is when handling the LDC opcode (so not in this function). In that case,  If we request the code for a contract starting at offset 0 then we can import directly into the stack. In fact, we can do this even if we request to load the contract at a greater offset than 0, but the logic becomes slightly complicated and error-prone. See https://github.com/FuelLabs/fuel-vm/pull/847/commits/ab0cc7bae7af3f97d4c8a9f2942a5834f7df8399 for how it was done.

Reply:
Ah I see, I misunderstood this and thought we talked about the stack for the rust code, not within the VM :sweat_smile:.